app/knights_dict/knights.py

```diff
@@ -0,0 +1,34 @@
+class Knight:
+    def __init__(self, knights: dict) -> None:
+        self.knights = knights
+        self.name = knights["name"]
+        self.power = knights["power"]
+        self.hp = knights["hp"]
+        self.protection = 0
+
+    def apply_weapon(self) -> None:
+        weapon_pow = self.knights["weapon"]["power"]
+        if isinstance(weapon_pow, int):
+            self.power += weapon_pow
+
+    def apply_armour(self) -> None:
+        self.protection = \
+            sum(armour["protection"] for armour in self.knights["armour"])

```

Comment:
Certainly, you can use parentheses to avoid using a backslash for line continuation